applications/vanilla/models/class.categorymodel.php

```diff
@@ -2801,4 +2805,126 @@ public function recalculateTree() {
     public static function flattenTree($categories) {
         return self::instance()->collection->flattenTree($categories);
     }
+
+    /**
+     * Adjust the aggregate post counts for a category.
+     *
+     * @param int|string|object|array $category
+     * @param int $offset
+     */
+    private static function adjustAggregateCounts($category, $offset) {
+        $categoryID = false;
+        $offset = intval($offset);
+
+        if (is_numeric($category)) {
+            $categoryID = $category;
+        } elseif (is_string($category)) {
+            $category = self::instance()->collection->get($category);
+        }
+
+        if (is_array($category) || is_object($category)) {
+            $categoryID = val('CategoryID', $category);
+        }
+
+        if (empty($categoryID)) {
+            return;
+        }
```

Comment:
These above 10 lines feel like a disambiguation method that other bits of code could make use of to just feed this method a reliable piece of data (CategoryID?) rather than literally accepting 4 variable types in every method that we need to reference a category. Making the var type `int|string|object|array` is basically making it `¯\_(ツ)_/¯`.